src/mscorlib/src/System/String.cs

```diff
@@ -3449,36 +3436,37 @@ public static string Concat(params object[] args)
 
 
         [System.Security.SecuritySafeCritical]  // auto-generated
-        public static String Concat(String str0, String str1) {
+        public unsafe static String Concat(String str0, String str1) {
             Contract.Ensures(Contract.Result<String>() != null);
             Contract.Ensures(Contract.Result<String>().Length ==
                 (str0 == null ? 0 : str0.Length) +
                 (str1 == null ? 0 : str1.Length));
             Contract.EndContractBlock();
 
-            if (IsNullOrEmpty(str0)) {
-                if (IsNullOrEmpty(str1)) {
-                    return String.Empty;
-                }
-                return str1;
+            if (IsNullOrEmpty(str0))
+            {
+                return str1 ?? string.Empty;
             }
 
-            if (IsNullOrEmpty(str1)) {
+            if (IsNullOrEmpty(str1))
+            {
                 return str0;
             }
 
-            int str0Length = str0.Length;
-            
-            String result = FastAllocateString(str0Length + str1.Length);
-            
-            FillStringChecked(result, 0,        str0);
-            FillStringChecked(result, str0Length, str1);
+            int totalLength = str0.Length + str1.Length;
+            string result = FastAllocateString(totalLength);
```

Comment:
I agree that this is simple enough ... I was more just trying to point out how easy it is to miss buffer overrun in unsafe code, even in very simple code.